community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1461,6 +1461,18 @@ class ReservationDetails:
     policies: List[str] # names (not URLs) of resource policies
     bulk_insert_name: str # name in format suitable for bulk insert (currently identical to user supplied name in long format)
 
+@dataclass(frozen=True)
+class FutureReservation:
+    project: str
+    zone: str
+    name: str
+    active: Optional[bool]
```

Comment:
Is it possible to have `active != (active_reservation != None)`?
If not, than it's pure derivative, consider adding 

```py
@property 
def active(self) -> bool:
  return self.active_reservation != None
```

Reply:
realized `active`  and `active_reservation` both track essentially the same thing. So `active` has been dropped.